Fix test_other_col_types when running in time zones with a positive offset from UTC

0xced opened the pull request:
This test would only pass when run in time zones with a negative offset from UTC. Since the test only asserts on year, month and day, the test passes. When run in a timezone with a positive offset from UTC, the day would change. For example, in UTC+2, the date would become 1968-03-14T22:00:00.0000000 and the test would fail (14 != 15).

By doing `d.ToUniversalTime()` the time of the supplied DateTime was changed and the stored date was actually altered (by its offset from UTC). Asserting on the whole date would fail:
>    Assert.Equal() Failure  
> Expected: 1968-03-15T00:00:00.0000000  
> Actual:   1968-03-15T08:00:00.0000000  

The assertion has been changed to assert on the full DateTime and the code was fixed to not alter the supplied date. So now, `test_other_col_types` always passes, no matter what `TimeZoneInfo.Local` is.

ericsink commented:
The breaking change to ugly.cs is problematic.

The intent of that code is to convert the given DateTime to a "unix timestamp", which is defined in terms of UTC.

0xced commented:
Indeed, even if *ugly* is not meant for production, a breaking change is never nice. So I reverted the change in my last commit.

Instead, I changed the DateTime which is returned when querying to be explicitly of [kind](https://docs.microsoft.com/en-us/dotnet/api/system.datetimekind) **Utc** and adapted the test accordingly.

ericsink commented:
"even if ugly is not meant for production"

FWIW, SQLitePCLRaw.ugly *is* used in production.  My coworkers and I (at SourceGear) use it in a variety of projects.

0xced commented:
Maybe *not meant for production* was a bit of a stretch, I extrapolated from the ugly class documentation:

> It retains all the "lower-case and underscores" ugliness of the layer below.  As such, this is not a wrapper intended for public consumption.

Anyway, the current pull request should not introduce a breaking change. It just returns a DateTime with kind `DateTimeKind.Utc` instead of `DateTimeKind.Unspecified`.

ericsink commented:
I understand.  I was just clarifying.
